gen/src/generate/ucd/shared/version.rs

```diff
@@ -0,0 +1,57 @@
+use std::fs::File;
+use std::io::{self, Read, Write};
+use std::path::Path;
+use std::str::FromStr;
+
+use generate::PREAMBLE;
+
+use regex::Regex;
+
+pub struct UnicodeVersion(u16, u16, u16);
```

Comment:
It could make sense to move `UnicodeVersion` up, but not on the case of the table generation code. I personally think that the table generation should be completely separate from the actual library. The version struct(s) will probably be the only duplication anyway, as all the other data structures in the generation code are more likely to reflect the data source than target.